src/dotnet/src/Mil.Navy.Nrl.Norm/NormFile.cs

```diff
@@ -29,16 +26,14 @@ public unsafe string Name
         {
             get
             {
-                var buffer = stackalloc byte[FILENAME_MAX];
-                var bufferPtr = (nint)buffer;
+                var buffer = stackalloc sbyte[FILENAME_MAX];
 
-                if (!NormFileGetName(_handle, bufferPtr, FILENAME_MAX))
+                if (!NormFileGetName(_handle, buffer, FILENAME_MAX))
                 {
                     throw new IOException("Failed to get file name");
                 }
-                var name = Marshal.PtrToStringAnsi(bufferPtr);
-
-                return name ?? string.Empty;
+                
+                return new string(buffer);
```

Comment:
Nice, didn't notice this constructor. It's not exactly ASCII, but it will probably do. It's also possible to specify the encoding, if needed.